src/main/java/com/joss/conductor/mobile/Locomotive.java

```diff
@@ -572,19 +572,23 @@ private Locomotive performCornerSwipe(ScreenCorner corner, SwipeElementDirection
     }
 
     public WebElement swipeTo(SwipeElementDirection s, By by, int attempts) {
-        int i;
-
-        if (isPresentWait(by)) {
-            return driver.findElement(by);
-        } else {
-            for (i = 0; i < attempts; i++) {
-                swipeCenterLong(s);
-                if (isPresent(by)) {
-                    return driver.findElement(by);
+        WebElement element;
+        for (int i = 0; i < attempts; i++) {
+            swipeCenterLong(s);
+            try {
+                element = driver.findElement(by);
+                // element was found, check for visibility
+                if (element.isDisplayed()) {
+                    // element is in view, exit the loop
+                    return element;
                 }
+                // element was not visible, continue scrolling
+            } catch (org.openqa.selenium.NoSuchElementException e) {
```

Comment:
can we do an import at the beginning of the file for the exception? `import org.openqa.selenium.NoSuchElementException`?
Also let's not abbreviate variable, lets change `e` to `exception` or `ignored`